packages/nimble-components/src/select/specs/built-in-labels.md

```diff
@@ -0,0 +1,77 @@
+# Built-in Labels for Select and Combobox
+
+## Problem Statement
+
+Most Nimble controls offer built in labels but the select and combobox controls don't, meaning clients have had to create and style labels manually.
+
+Built-in labels will reduce the effort needed for clients to consume these controls, as well as resolving some accessibility issues (see work item links below).
+
+## Links To Relevant Work Items and Reference Material
+
+[Built-in label for select and combobox](https://github.com/ni/nimble/issues/2183)
+
+Related issues:
+
+-   [Resolve label accessibility issues](https://github.com/ni/nimble/issues/94)
+-   [Fix accessibility issues in Angular Example app](https://github.com/ni/nimble/issues/280)
+
+## Implementation / Design
+
+Implementation details apply to both `nimble-select` and `nimble-combobox`, unless otherwise specified.
+
+### API (Slots)
+
+Default Slot:
+
+-   The `nimble-list-option`s in the control are placed in this slot currently.
+-   We will add support for specifying the label as text content in the same slot. For example:
+    ```html
+    <nimble-select>
+        Select Label Text
+        <nimble-list-option>A</nimble-list-option>
+        <nimble-list-option>B</nimble-list-option>
+    </nimble-select>
+    ```
+
+### Templates / Styling
```

Comment:
A couple layout considerations:
1. if the label is not provided then should vertical space be reserved for it? I presume not, since that will maintain layout compatibility for existing clients and that matches what other labels do.
2. if the label doesn't fit in the width of the control what will happen? For our other components with labels it looks like they grow to fit the label.

Reply:
Good points, added to the HLD. (1 - no reserved space if label not present; 2- said we'll probably grow to fit like the other controls)